pkg/messages/baseMessage.go

```diff
@@ -52,23 +57,26 @@ type StateMessage interface {
 type RequestMessage interface {
 	Message
 	Request() int
-	Reply(senderFloor int, response bool) ResponseMessage
+	Reply(senderFloor int, response bool, returnId uuid.UUID) ResponseMessage
```

Comment:
I'm finding it hard to get my head around this, could someone explain? Is this basically saying "a message of type request message has to implement a plausible response message that can be used to reply back"? If so, shouldn't that just be a message type? Or would you do `Message.Reply`? I'm a bit confused what's the logic here

Reply:
I think the variable name `returnId` could be improved... just not sure what for.

The `Reply()` function is called by the agent which receives the `RequestMessage()`, and this then creates the reply message. `returnId` is the id of the message which the agent is replying to, i.e. the `RequestMessage`s UUID.

Reply:
(I think)

Reply:
I'm still confused, sorry. So the agent receives a message `receivedMessage` in their handler, what is `receivedMessage.Reply(...)` doing here? Does it create a response message that I can then send? If so does it mean it can only create a response message of type bool?

Reply:
so the agent would send back `receivedMessage.Replay(a.Floor(), RESPONSE, ?????)`? what's `returnId` here?

Reply:
I think I get it. I think there is duplication here. You are already calling `Reply` on the message, and then you are also passing in the `returnId` and using it. Shouldn't it be either? Since you're calling `Reply` on the message you should be able to pull the message id directly (the whole point of this convo implementation is to reduce implementation complexity so agent doesn't need to be aware of both the ID AND the reply mechanism)

Reply:
Resolved in private convo, thanks Jason